modules/platform/nuxeo-platform-document-routing/nuxeo-routing-core/src/main/java/org/nuxeo/ecm/platform/routing/core/listener/RouteModelsInitializator.java

```diff
@@ -32,13 +40,43 @@
  */
 public class RouteModelsInitializator extends RepositoryInitializationHandler {
 
+    /** @since 11.5 */
+    public static final String CLUSTER_START_DURATION_PROP = "org.nuxeo.workflow.cluster.start.duration";
+
     @Override
     public void doInitializeRepository(CoreSession session) {
-        // This method gets called as a system user
-        // so we have all needed rights to do the check and the creation
+        String repositoryName = session.getRepositoryName();
+        Duration duration = getLockDuration();
+        Duration pollDelay = Duration.ofSeconds(1);
+        ClusterService clusterService = Framework.getService(ClusterService.class);
+        clusterService.runAtomically("start-workflows-" + repositoryName, duration, pollDelay,
+                () -> importWorkflows(repositoryName));
+    }
+
+    protected void importWorkflows(String repositoryName) {
         DocumentRoutingService service = Framework.getService(DocumentRoutingService.class);
-        service.importAllRouteModels(session);
-        session.save();
+        CoreInstance.doPrivileged(repositoryName, session -> {
+            service.importAllRouteModels(session);
+            session.save();
+        });
+    }
+
+    protected Duration getLockDuration() {
+        Duration duration;
+        try {
+            String prop = Framework.getProperty(CLUSTER_START_DURATION_PROP);
+            if (isBlank(prop)) {
+                duration = Duration.ZERO;
```

Comment:
What do you think about adding the `isBlank` check to `DurationUtils` and throw a DateTimeParseException in such a case?

Reply:
Actually I already added `DurationUtils.parsePositive` specifically for that in the past, see  https://github.com/nuxeo/nuxeo/commit/99c78f67c5bb4eca7f43743359f074c6f257790d but I don't know why I copy/paste old code in this case...